bika/lims/adapters/referencewidgetvocabulary.py

```diff
@@ -33,34 +34,72 @@ def __call__(self, result=None, specification=None, **kwargs):
         # first with all queries
         contentFilter = dict((k, v) for k, v in base_query.items())
         contentFilter.update(search_query)
-        try:
+
+        # Sorted by? (by default, Title)
+        sort_on = self.request.get('sidx', 'Title')
+        if sort_on == 'Title':
+            sort_on = 'sortable_title'
+        if sort_on:
+            # Check if is an index and if is sortable. Otherwise, assume the
+            # sorting must be done manually
+            index = catalog.Indexes.get(sort_on, None)
+            if index and index.meta_type in ['FieldIndex', 'DateIndex']:
+                contentFilter['sort_on'] = sort_on
+                # Sort order?
+                sort_order = self.request.get('sord', 'asc')
+                if (sort_order in ['desc', 'reverse', 'rev', 'descending']):
+                    contentFilter['sort_order'] = 'descending'
+                else:
+                    contentFilter['sort_order'] = 'ascending'
+
+        # Can do a search for indexes?
+        criterias = []
+        fields_wo_index = []
+        if searchTerm:
+            for field_name in searchFields:
+                index = catalog.Indexes.get(field_name, None)
+                if not index:
+                    fields_wo_index.append(field_name)
+                if index.meta_type in ('ZCTextIndex'):
+                    searchTerm += '*'
```

Comment:
Also better if we do not add this '*' to searchTerm but save it in another temporary variable? Imagine we have 2 'ZCTextIndex' and a 'FieldIndex'. Then when adding 'FieldIndex' search criteria, searchTerm will have 2 asterisks trailing on original value and since 'FieldIndex' doesn't accept it as Regular Expression, it will not work for that 'FieldIndex'.